webui/react/src/components/RegisterModelVersionModal.tsx

```diff
@@ -0,0 +1,203 @@
+import { Input, Modal, notification, Select } from 'antd';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import handleError, { ErrorType } from 'ErrorHandler';
+import { paths } from 'routes/utils';
+import { getModels, postModelVersion } from 'services/api';
+import { V1GetModelsRequestSortBy } from 'services/api-ts-sdk';
+import { validateDetApiEnum } from 'services/utils';
+import { Metadata, ModelItem } from 'types';
+import { isEqual } from 'utils/data';
+
+import Link from './Link';
+import EditableMetadata from './Metadata/EditableMetadata';
+import NewModelModal from './NewModelModal';
+import css from './RegisterModelVersionModal.module.scss';
+import EditableTagList from './TagList';
+
+const { Option } = Select;
+
+interface Props {
+  checkpointUuid: string;
+  onClose?: () => void;
+  onCloseAll?: () => void;
+  visible?: boolean;
+}
+
+const RegisterModelVersionModal: React.FC<Props> = (
+  { checkpointUuid, visible = false, onClose, onCloseAll },
+) => {
+  const [ selectedModelId, setSelectedModelId ] = useState<number>();
+  const [ models, setModels ] = useState<ModelItem[]>([]);
+  const [ versionName, setVersionName ] = useState('');
+  const [ versionDescription, setVersionDescription ] = useState('');
+  const [ tags, setTags ] = useState<string[]>([]);
+  const [ metadata, setMetadata ] = useState<Metadata>({});
+  const [ expandDetails, setExpandDetails ] = useState(false);
+  const [ canceler ] = useState(new AbortController());
+  const [ showNewModelModal, setShowNewModelModal ] = useState(false);
+
+  const fetchModels = useCallback(async () => {
+    try {
+      const response = await getModels({
+        orderBy: 'ORDER_BY_DESC',
+        sortBy: validateDetApiEnum(
+          V1GetModelsRequestSortBy,
+          V1GetModelsRequestSortBy.LASTUPDATEDTIME,
+        ),
+      }, { signal: canceler.signal });
+      setModels(prev => {
+        if (isEqual(prev, response.models)) return prev;
+        return response.models;
+      });
+    } catch(e) {
+      handleError({ message: 'Unable to fetch models.', silent: true, type: ErrorType.Api });
+    }
+  }, [ canceler.signal ]);
+
+  const registerModelVersion = useCallback(async () => {
+    if (!selectedModelId) return;
+    try {
+      const response = await postModelVersion({
+        body: {
+          checkpointUuid,
+          comment: versionDescription,
+          labels: tags,
+          metadata,
+          modelId: selectedModelId,
+          name: versionName,
+        },
+        modelId: selectedModelId,
+      });
+      onCloseAll?.();
+      notification.open(
+        {
+          btn: null,
+          description: (
+            <div className={css.toast}>
+              <p>{`"${versionName || `Version ${selectedModelNumVersions + 1}`}"`} registered</p>
+              <Link path={paths.modelVersionDetails(selectedModelId, response?.id ?? 0)}>
```

Comment:
I have two suggestions on this notification - more specific text ("View Model Version") and handling a missing response.id differently (maybe as an error?)